tests/mock_virtual_machine.h

```diff
@@ -66,7 +66,13 @@ struct MockVirtualMachineT : public T
     MOCK_METHOD(std::string, management_ipv4, (), (override));
     MOCK_METHOD(std::vector<std::string>, get_all_ipv4, (), (override));
     MOCK_METHOD(std::string, ipv6, (), (override));
-    MOCK_METHOD(std::string, ssh_exec, (const std::string& cmd), (override));
+
+    MOCK_METHOD(std::string, ssh_exec_mock, (const std::string& cmd, bool whisper));
+    std::string ssh_exec(const std::string& cmd, bool whisper = false) override
```

Comment:
Can you explain to me the purpose of this extra implementation of `ssh_exec()` and why you've added `ssh_exec_mock` instead of just expanding the argument list?

Reply:
Hmm good thing you highlighted this, I don't think I was thinking straight... 

Google mock doesn't support having a default argument in `MOCK_METHOD` calls and normally, when overriding a virtual method with a default argument, one should make sure that the same default is used. Otherwise, different defaults could take effect depending on the concrete type of the object that the method is called on. Or it may be impossible to call the default version. When mocking, I suppose it would be OK to just ignore the default argument... In this case we call the method directly via the mock in a couple of places: https://github.com/canonical/multipass/blob/6719804fb179062cee288b1a9bdd50e675f34622/tests/test_base_virtual_machine.cpp#L1372)

But I guess we could just add a boolean to the call. Or we could add an ssh_exec without the second argument. But it may be better to avoid the default argument entirely to begin with. Let me revisit this.

Reply:
Thanks for calling my attention to this @sharder996, it should be better now.

Reply:
Thanks, It looks much better now!

The private side will need to be updated as well.

Reply:
Ah indeed.

Reply:
Actually, I don't think we even need a private side anymore, do we?

Reply:
Right, since it was just for a function name change.